lib/util/compile.js

```diff
@@ -52,6 +52,14 @@ function which_node_gyp() {
     if (existsSync(node_gyp_bin)) {
         return node_gyp_bin;
     }
+
+    try {
+        var maybe_elsewhere = path.join(path.dirname(fs.realpathSync(which('npm'))), 
```

Comment:
Looks like you forgot to require `which`? Also, is it absolutely needed? Would be great to not add more deps.